### DISCUSSION TITLE
mocp -a may abort due to unexpected events if playback is active

### DISCUSSION THREAD

**Discussion by dreamlayers** [jonsafari/mocp#41] [General]
Adding a multiple folders with one album in each via `mocp -a`  on a cifs filesystem mount in Ubuntu 24.04 sometimes aborts at: https://github.com/jonsafari/mocp/blob/48b48e593690ef0c398d5e33dc119b0a4fc73a2b/interface.c#L1248-L1249

In the gdb backtrace, selecting this stack frame and doing `info locals` shows `type = 2`: https://github.com/jonsafari/mocp/blob/48b48e593690ef0c398d5e33dc119b0a4fc73a2b/protocol.h#L48

I guess the issue happens when such other events happen while that function is executing.  If the function takes a longer time to run because many files are being added and/or the filesystem they're on is slow, this increases chances of this happening.

First stopping playback via `mocp -s` seems to prevent this problem. When I was setting a new playlist via `mocp -sc` followed by `mocp -ap` I didn't get any aborts. 

Maybe this should be posted at the bug report forum at https://moc.daper.net/forum/19 but I don't see how to register on the forums over there.

**Comment by tomaszg7** [comment]
> Maybe this should be posted at the bug report forum at https://moc.daper.net/forum/19 but I don't see how to register on the forums over there.

Account creation was temporarily suspended to cut down on spam. It is now active again, feel free to post this report on the MOC website.

BTW. I use MOC almost exclusively on NFS (over LAN) and never had similar problem, interesting.
